src/Roave/SecurityAdvisories/AdvisorySources/GetAdvisoriesFromGithubApi.php

```diff
@@ -104,11 +102,13 @@ static function (array $item) {
      * that is used to do a sequence of "paged" requests.
      * Note: 'endCursor' contains the least cursor in the given batch
      *
-     * @return Advisory[]
+     * @return mixed[]
```

Comment:
that is where psalm was complaining about wrong return type. Instead of a list of Advisories we return here an array of data from GitHub

Reply:
Can we define a better structure here?

Reply:
sure, I'll try to go with ValueObject here if that is okay?
But it will yield a lot of changes :thinking: 
`\Roave\SecurityAdvisories\Advisory::fromArrayData(array $config)` will have to be changed to smth like `\Roave\SecurityAdvisories\Advisory::from(AdvisoryConfig $config)`

Reply:
`@psalm-return array{foo: type, bar: type}` would suffice

Reply:
done. Although I'm not sure about formatting, I tried something like this
```
     * @psalm-return array<int, array{
     *      cursor: string,
     *      node: array{
     *          vulnerableVersionRange: string,
     *          package: array{name: string}
     *      }
     * }>
```
looks ugly to me :unamused:

Reply:
PHP types are ugly: they aren't spelled out :)